pandas/core/internals.py

```diff
@@ -1854,11 +1857,30 @@ class ExtensionBlock(NonConsolidatableMixIn, Block):
 
     ExtensionArrays are limited to 1-D.
     """
+
+    def __init__(self, values, placement, ndim=None):
+        values = self._maybe_coerce_values(values)
+        super(ExtensionBlock, self).__init__(values, placement, ndim)
+
+    def _maybe_coerce_values(self, values):
+        # Unboxes Series / Index
+        # Doesn't change any underlying dtypes.
```

Comment:
why is this not just ``_as_best_array()`` (or new name), this kind of if logic is polluting (yes its all over the place, a major goal of this ExtensionArray is to clean this up, I would argue this is THE goal).

Reply:
can you make a real doc-string

Reply:
Good point, since there's no parent.